cranelift/codegen/src/isa/riscv64/abi.rs

```diff
@@ -592,20 +630,24 @@ impl ABIMachineSpec for Riscv64MachineDeps {
     }
 
     fn get_regs_clobbered_by_call(_call_conv_of_callee: isa::CallConv) -> PRegSet {
-        let mut v = PRegSet::empty();
-        for (k, need_save) in CALLER_SAVE_X_REG.iter().enumerate() {
-            if !*need_save {
-                continue;
-            }
-            v.add(px_reg(k));
-        }
-        for (k, need_save) in CALLER_SAVE_F_REG.iter().enumerate() {
-            if !*need_save {
-                continue;
-            }
-            v.add(pf_reg(k));
+        let x_clobbers = CALLER_SAVE_X_REG
+            .iter()
+            .enumerate()
+            .filter(|(_, clobbered)| **clobbered)
+            .map(|(i, _)| px_reg(i));
+        let f_clobbers = CALLER_SAVE_F_REG
+            .iter()
+            .enumerate()
+            .filter(|(_, clobbered)| **clobbered)
+            .map(|(i, _)| pf_reg(i));
+        // We clobber all vector registers (and vector state) on calls.
+        let v_clobbers = (0..=31).map(|i| pv_reg(i));
+
+        let mut clobbers = PRegSet::empty();
+        for clobber in x_clobbers.chain(f_clobbers).chain(v_clobbers) {
+            clobbers.add(clobber);
         }
-        v
+        clobbers
```

Comment:
This implementation looks correct, but FWIW the *hope* at least with the `PRegSet` definition was to allow for constants (they're small bitsets, u128 or now u256) to be returned by backends. I realize the existing backend also doesn't do this, so in the worst case we can keep the status quo. But this might even be a small but visible performance issue for function bodies with frequent calls. Would it be possible to define a `const fn` that computes a `PRegSet` here (either from existing constants or with a straight-line "add reg ..." sequence)?